### PR TITLE
feat(kuma-cp) Support for gRPC

### LINKED ISSUE #912: Support gRPC protocol
### Summary

Kuma currently supports HTTP/1.1 and HTTP/2 (after https://github.com/kumahq/kuma/pull/911 is merged).
We should introduce gRPC support in Kuma. This means when user put `grpc` in `protocol` (described here https://kuma.io/docs/0.6.0/policies/http-support-in-kuma/ ) we configure resources just like for http2 (gRPC is based on HTTP/2) but we also add an extra HTTP filter on HTTP Connection Manager that emits gRPC stats (https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/filter/http/grpc_stats/v2alpha/config.proto)

### PR DESCRIPTION
**Pull request by tharun208**
Signed-off-by: Tharun <rajendrantharun@live.com>

### Summary
Add gRPC Support

Fixes #912

### Documentation
[https://github.com/kumahq/kuma-website/pull/254](https://github.com/kumahq/kuma-website/pull/254)

### CODE REVIEW

File: pkg/xds/generator/protocol.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ var (
 	// GRPC has a protocol stack [GRPC, HTTP2, TCP],
 	// TCP  has a protocol stack [TCP].
 	protocolStacks = map[mesh_core.Protocol]mesh_core.ProtocolList{
+		mesh_core.ProtocolGRPC:  {mesh_core.ProtocolGRPC, mesh_core.ProtocolHTTP2, mesh_core.ProtocolTCP},
```
[the remark this answers is not in the document]
Comment (tharun208) [reply pkg/xds/generator/protocol.go:15]: yes

[the remark this answers is not in the document]
Comment (nickolaev) [reply pkg/xds/generator/protocol.go:15]: So you need to rewrite it like this:
```golang
		mesh_core.ProtocolGRPC:  {mesh_core.ProtocolGRPC, mesh_core.ProtocolHTTP2, mesh_core.ProtocolHTTP, mesh_core.ProtocolTCP},
```

Or am I getting it wrong?

[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply pkg/xds/generator/protocol.go:15]: Actually it should be
```
mesh_core.ProtocolGRPC:  {mesh_core.ProtocolGRPC, mesh_core.ProtocolHTTP2, mesh_core.ProtocolTCP},
mesh_core.ProtocolHTTP2:  {mesh_core.ProtocolHTTP2, mesh_core.ProtocolTCP},
```
can you fix HTTP2 @tharun208 ? Thanks.

Configuring HTTP2 app with HTTP is a problem, but TCP will still work.

**Comment by jakubdyszkiewicz** [comment]
I'd love to introduce E2E test for this with a real gRPC app, but this might be a little bit difficult to set this up right now.

Have you tested it manually spawning client -> envoy -> server, whereas the client uses gRPC to connect to the server?

**Comment by tharun208** [comment]
> I'd love to introduce E2E test for this with a real gRPC app, but this might be a little bit difficult to set this up right now.
> 
> Have you tested it manually spawning client -> envoy -> server, whereas the client uses gRPC to connect to the server?
yes. It is working fine. was able to a .response from server.

**Comment by tharun208** [comment]
> I'd love to introduce E2E test for this with a real gRPC app, but this might be a little bit difficult to set this up right now.
> 
> Have you tested it manually spawning client -> envoy -> server, whereas the client uses gRPC to connect to the server?

Yes. I tested with a grpc application and able to get a response from the server.

**Comment by jakubdyszkiewicz** [comment]
Awesome, can you also add docs in kuma-website? Otherwise, users won't know that we now support gRPC.

**Pull request merged**